numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1200,15 +1200,18 @@ _void_compare(PyArrayObject *self, PyArrayObject *other, int cmp_op)
             }
         }
         if (res == NULL && !PyErr_Occurred()) {
-            PyErr_SetString(PyExc_ValueError, "No fields found.");
+            /* these dtypes had no fields */
+            res = PyArray_NewLikeArray(self, NPY_ANYORDER,
```

Comment:
This doesn't broadcast with `other`, unless the broadcasting has already happened here

Reply:
Ah, yeah. I think I've got it finally by using a multiiter.